Assets/Scripts/Actors/ActorComponents/ActorPhysics.cs

```diff
@@ -497,36 +426,30 @@ void InitializeStateMethodMap()
 	{
 		foreach ( ActorStates state in System.Enum.GetValues( typeof( ActorStates ) ) )
 		{
-			if ( !_stateMethodMap.ContainsKey( state ) )
+			if ( !_stateMap.ContainsKey( state ) )
 			{
-				_stateMethodMap.Add( state, delegate() { } );
+				_stateMap.Add( state, new DefaultState() );
 			}
 		}
 	}
 
-	void ChangeState( ActorStates toState )
+	/**
+	 * Transition between physics states.
+	 * 
+	 * Note: This immediately calls the Exit() method of the
+	 * old state and then the Enter() method of the new state.
+	 * If you call ChangeState() multiple times in one frame
+	 * the state transition will take place multiple times,
+	 * including whatever side effects that may entail.
+	 * Best practices is to ensure that you only transition
+	 * states at most once per frame.
+	 */
+	public void ChangeState( ActorStates toState )
 	{
-		_currentState = toState;
-		CurrentStateMethod = _stateMethodMap[_currentState];
-	}
-
-	void StartClimbing( ClimbableTag climbTag )
-	{
-		_climbTag = climbTag;
-		_climbSurface = _climbTag.GetComponent<Transform>();
-
-		ChangeState( ActorStates.Climbing );
-		
-		rigidbody.useGravity = false;
-		rigidbody.velocity = Vector3.zero;
-		lifter.gameObject.SetActive( false );
-		bumper.gameObject.SetActive( false );
-		climbBumper.gameObject.SetActive( true );
-
-		if ( !actor.animator )
-		{
-			actor.animator.SetBool( "isClimbing", true );
-		}
+		_currentStateType = toState;
+		_currentState.Exit();
+		_currentState = _stateMap[_currentStateType];
```

Comment:
personally I'd replace the _currentStateType here with toState just in case the order of these statements ever gets switched around.